static/js/settings.js

```diff
@@ -152,22 +152,140 @@ const SettingsScreen = {
+    addDomainLocalButton.addEventListener('click', () => {
+      this.onLocalDomainClick();
+    });
+
+    // Comented out until full integration of Dynamic tunnel creation
+    // with Service Discovery
+    // const addDomainTunnelButton =
+    //   document.getElementById('settings-domain-moz-tunnel-change');
+    //     addDomainTunnelButton.addEventListener('click', () => {
+    //     this.onTunnelDomainClick();
+    // });
+
     fetch('/settings/tunnelinfo', opts).then(function(response) {
-      return response.text();
+      return response.json();
     }).then(function(body) {
       if (body) {
-        document.getElementById('current-domain').innerText = body;
+        document.getElementById('domain-settings-local-name')
+          .value = body.localDomain;
+        document.getElementById('domain-settings-local-checkbox')
+          .checked = body.mDNSstate;
+        document.getElementById('domain-settings-tunnel-name')
+          .innerText = body.tunnelDomain;
       } else {
-        document.getElementById('current-domain').innerText = 'Not set.';
+        document.getElementById('domain-settings-local-name')
+          .value = 'Unknown state.';
+        document.getElementById('domain-settings-tunnel-name')
+          .value = 'Unknown state.';
       }
     });
   },
 
+
+  // The button controller to update the local domain settings.
+  // In menu -> Settings -> Domain
+
+  onLocalDomainClick: function() {
+    const localDomainCheckbox = document.getElementById(
+      'domain-settings-local-checkbox');
+    const localDomainName = document
+      .getElementById('domain-settings-local-name');
+    const error = document.getElementById('domain-settings-error');
+    const data = {local: {multicastDNSstate: localDomainCheckbox.checked,
+                          localDNSname: localDomainName.value}};
+
+    const opts = {
+      method: 'PUT',
+      body: JSON.stringify(data),
+      headers: {
+        Authorization: `Bearer ${window.API.jwt}`,
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+    };
+
+
+    fetch('/settings/domain', opts)
+      .then(function(response) {
+        return response.json();
+      })
+      .then(function(domainJson) {
+        // if the update was successful, we have a legit local domain and mDNS
+        // is active then redirect
+        if (domainJson.update && domainJson.localDomain.length > 0) {
+          if (domainJson.mDNSstate) {
+            window.location.href = domainJson.localDomain;
+          } else {
+            // the server says mDNS is switched off so redirect back to
+            // localhost
+            window.location.href = 'https://localhost:4443/';
+          }
+        } else {
+          error.classList.remove('hidden');
+          error.textContent = domainJson.error;
+          document.getElementById('domain-settings-local-name')
+            .value = domainJson.localDomain;
+
```

Comment:
`eslint` issue: remove blank line before end of block.

Reply:
done.